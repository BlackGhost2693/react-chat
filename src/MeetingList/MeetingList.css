.rce-container-mtlist {
    display: block;
    overflow: auto;
}
