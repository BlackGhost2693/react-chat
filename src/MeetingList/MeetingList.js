import React from 'react';
import './MeetingList.css';

import MeetingItem from '../MeetingItem/MeetingItem';

const classNames = require('classnames');

function MeetingList(props) {

  const onClick = (item, i, e) => {
    if (props.onClick instanceof Function)
      props.onClick(item, i, e);
  }

  const onContextMenu = (item, i, e) => {
    e.preventDefault();
    if (props.onContextMenu instanceof Function)
      props.onContextMenu(item, i, e);
  }

  const onAvatarError = (item, i, e) => {
    if (props.onAvatarError instanceof Function)
      props.onAvatarError(item, i, e);
  }

  const onMeetingClick = (item, i, e) => {
    if (props.onMeetingClick instanceof Function)
      props.onMeetingClick(item, i, e);
  }

  const onShareClick = (item, i, e) => {
    if (props.onShareClick instanceof Function)
      props.onShareClick(item, i, e);
  }

  const onCloseClick = (item, i, e) => {
    if (props.onCloseClick instanceof Function)
      props.onCloseClick(item, i, e);
  }

  return (
    <div
      ref={props.cmpRef}
      className={classNames('rce-container-mtlist', props.className)}>
      {
        props.dataSource.map((x, i) => (
          <MeetingItem
            id={x.id || i}
            key={i}
            lazyLoadingImage={props.lazyLoadingImage}
            {...x}
            onAvatarError={(e) => onAvatarError(x, i, e)}
            onContextMenu={(e) => onContextMenu(x, i, e)}
            onClick={(e) => onClick(x, i, e)}
            onMeetingClick={(e) => onMeetingClick(x, i, e)}
            onShareClick={(e) => onShareClick(x, i, e)}
            onCloseClick={(e) => onCloseClick(x, i, e)}/>
        ))
      }
    </div>
  );
}

MeetingList.defaultProps = {
  dataSource: [],
  onClick: null,
  lazyLoadingImage: undefined,
};

export default MeetingList;
