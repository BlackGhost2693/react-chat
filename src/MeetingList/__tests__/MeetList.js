import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MeetingList from '../MeetingList';

describe('MeetingList component', () => {
  it('should render without issues', () => {
    const component = shallow(<MeetingList />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
