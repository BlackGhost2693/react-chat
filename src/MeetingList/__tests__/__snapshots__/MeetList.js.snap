// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingList component should render without issues 1`] = `
<div
  className="rce-container-mtlist"
/>
`;
